.github/workflows/ci.yml

```diff
@@ -1,25 +1,22 @@
 # GitHub Actions - CI for Go to build & test.  See ci-go-cover.yml and linters.yml for code coverage and linters.
 # Taken from: https://github.com/fxamacker/cbor/workflows/ci.yml (thanks!)
 name: ci
-on: [push]
+on: [push, pull_request]
 jobs:
 
   # Test on various OS with default Go version. 
   tests:
-    name: Test on ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.17.x, 1.18.x]
```

Comment:
Do we still need `1.17.x` since `go.mod` is now `1.18`?

Reply:
We still want to test against older versions, so we are sure those are supported.

The `go` directive in `go.mod` does not set the minimum supported Go version, one can still use older versions. It indicates to the compiler which Go features are allowed, i.e. `go 1.17` would mean fuzzing and generics are not available even if go1.18 compiler is used.

The `go` directive does help on keeping backwards compatibility with older versions, but there are other ways to achieve it. For example, our fuzz tests are placed in `fuzz_test.go`, which is gated behind the build constraint `//go:build go1.18`. Therefore go1.17 and older compilers will be able to test the module, but without running the fuzz tests.